include/exadg/incompressible_navier_stokes/user_interface/parameters.cpp

```diff
@@ -538,14 +535,43 @@ Parameters::check(dealii::ConditionalOStream const & pcout) const
                 dealii::ExcMessage("Not implemented."));
   }
 
-
   // TURBULENCE
-  if(use_turbulence_model)
+  if(turbulence_model_data.is_active)
   {
-    AssertThrow(turbulence_model != TurbulenceEddyViscosityModel::Undefined,
-                dealii::ExcMessage("parameter must be defined"));
-    AssertThrow(turbulence_model_constant > 0,
-                dealii::ExcMessage("parameter must be greater than zero"));
+    AssertThrow(turbulence_model_data.turbulence_model != TurbulenceEddyViscosityModel::Undefined,
+                dealii::ExcMessage("Parameter must be defined."));
+    AssertThrow(treatment_of_variable_viscosity != TreatmentOfVariableViscosity::Undefined,
+                dealii::ExcMessage("Parameter must be defined."));
+  }
+
+  // GENERALIZED NEWTONIAN MODEL
+  if(generalized_newtonian_model_data.is_active)
+  {
+    AssertThrow(generalized_newtonian_model_data.generalized_newtonian_model !=
+                  GeneralizedNewtonianViscosityModel::Undefined,
+                dealii::ExcMessage("Parameter must be defined."));
+    AssertThrow(treatment_of_variable_viscosity != TreatmentOfVariableViscosity::Undefined,
+                dealii::ExcMessage("Parameter must be defined."));
+  }
+
+  // VARIABLE VISCOSITY MODELS
+  if(viscosity_is_variable())
+  {
+    if(temporal_discretization == TemporalDiscretization::BDFDualSplittingScheme)
+    {
+      AssertThrow(
+        treatment_of_variable_viscosity == TreatmentOfVariableViscosity::Explicit,
+        dealii::ExcMessage(
+          "An implicit treatment of the variable viscosity field (rendering the viscous step of the dual splitting scheme nonlinear regarding the unknown velocity field) is currently not implemented for the dual splitting scheme."));
+    }
+
+    if(treatment_of_variable_viscosity == TreatmentOfVariableViscosity::Implicit)
+    {
+      AssertThrow(
+        implicit_convective_problem(),
+        dealii::ExcMessage(
+          "The current implementation only calls a Newton solver, if we have an implicit convective problem. Treat nonlinear convective and viscous terms alike."));
+    }
```

Comment:
Only if a Newton solver is called (depending _only_ on the convective term) do we also update the viscosity (using a fixed point algorithm during residual evaluation). Assert the case where the user calls for implcit viscosity treatment, but the convective term is treated explicitly. This will be addressed in a follow-up PR.